source/core/src/main/com/csse3200/game/components/player/AllFiringLogic.java

```diff
@@ -0,0 +1,141 @@
+package com.csse3200.game.components.player;
+
+import com.badlogic.gdx.audio.Sound;
+import com.badlogic.gdx.math.GridPoint2;
+import com.badlogic.gdx.math.Vector2;
+import com.csse3200.game.components.CombatStatsComponent;
+import com.csse3200.game.components.player.RangeDetectionComponent;
+import com.csse3200.game.components.player.WeaponComponent;
+import com.csse3200.game.components.projectile.ProjectileAttackComponent;
+import com.csse3200.game.entities.Entity;
+import com.csse3200.game.entities.configs.ProjectileConfig;
+import com.csse3200.game.entities.factories.ProjectileFactory;
+import com.csse3200.game.services.ServiceLocator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+
+
+
+public class AllFiringLogic {
```

Comment:
Since this class will be attached to the weapon entity, it should inherit from the Component class.